src/storage/src/hummock/sstable/multi_builder.rs

```diff
@@ -162,6 +182,29 @@ where
         if self.split_by_table && full_key.user_key.table_id.table_id != self.last_table_id {
             self.last_table_id = full_key.user_key.table_id.table_id;
             switch_builder = true;
+
+            if self.split_weight_by_vnode > 1 {
+                self.largest_vnode_in_current_partition =
+                    VirtualNode::COUNT / (self.split_weight_by_vnode as usize) - 1;
+            }
+        }
+        if self.largest_vnode_in_current_partition != VirtualNode::MAX.to_index() {
+            let key_vnode = full_key.user_key.get_vnode_id();
```

Comment:
The processing of each key will add an additional `get_vnode_id` call, we can verify whether a performance regression through the mirco bench.

We also need a benchmark to verify whether the modification is effective.

Reply:
> The processing of each key will add an additional `get_vnode_id` call, we can verify whether a performance regression through the mirco bench.
> 
> We also need a benchmark to verify whether the modification is effective.

That's why I put it in `if self.largest_vnode_in_current_partition != VirtualNode::MAX.to_index() {`
When `split_weight_by_vnode==0`, we do not need this `get_vnode_id` call.